lib/utils/form-builder/groupsHistory.ts

```diff
@@ -32,3 +33,51 @@ export const getPreviousIdFromCurrentId = (currentId: string, history: string[])
   }
   return null;
 };
+
+export const getRelevantValues = (
+  allFormValues: Responses,
+  allGroups: GroupsType | undefined,
+  answeredGroups: string[]
+) => {
+  if (!Array.isArray(answeredGroups) || !allGroups) return [] as unknown as Responses;
+  const relevantValues: Responses = {};
+  answeredGroups.forEach((answeredGroupId) => {
+    if (!allGroups[answeredGroupId] || !Array.isArray(allGroups[answeredGroupId].elements)) return;
+    const groupValueIntersection = allGroups[answeredGroupId].elements;
+    groupValueIntersection.forEach((relevantElement: string) => {
+      const key = relevantElement as keyof typeof Response;
+      const value = relevantElement as keyof typeof allFormValues;
+      relevantValues[key] = allFormValues[value];
+    });
+  });
+  return relevantValues;
+};
+
+export const filterNonRelevantValues = (allValues: Responses, valuesInHistory: Responses) => {
+  if (!allValues || !valuesInHistory) return {} as Responses;
+  const allValuesEmtpy = {} as Responses;
+  for (const key in allValues) {
+    allValuesEmtpy[key] = "";
+  }
+  return { ...allValuesEmtpy, ...valuesInHistory };
+};
+
+export const removeCustomFormValues = (values: Responses) => {
```

Comment:
Let's look to come up with a better name for this 

Maybe removeFormContext values

Reply:
Yup will change